src/OrchardCore.Modules/OrchardCore.Menu/Migrations.cs

```diff
@@ -17,7 +17,17 @@ public async Task<int> CreateAsync()
         {
             await _recipeMigrator.ExecuteAsync("menu.recipe.json", this);
 
-            return 1;
+            // Return 2 to shortcut the second migration on new content definition schemas.
+            return 2;
+        }
+
+        // Add content picker menu. This only needs to run on old content definition schemas.
+        // This code can be removed in a later version.
+        public async Task<int> UpdateFrom1Async()
+        {
+            await _recipeMigrator.ExecuteAsync("content-picker-menu-updatefrom1.recipe.json", this);
+
```

Comment:
@stantoxt what I have done here is two migrations, one for the new site setup, which has everything in it, and a second which will only run on old sites, with your new feature